openmdao/core/component.py

```diff
@@ -658,8 +660,14 @@ def declare_partials(self, of, wrt, dependent=True, rows=None, cols=None, val=No
             The type of approximation that should be used. Valid options include:
             'fd': Finite Difference, 'cs': Complex Step, 'exact': use the component
             defined analytic derivatives. Default is 'exact'.
-        **kwargs : dict
-            Keyword arguments for controlling the behavior of the approximation.
+        step : float
+            Step size for approximation. Default is the value of DEFAULT_FD_OPTIONS['step'].
```

Comment:
If this docstring is intended for the end user, I wonder if they should be told where to find `DEFAULT_FD_OPTIONS`?

Reply:
<img width="1076" alt="screen shot 2018-05-10 at 3 59 31 pm" src="https://user-images.githubusercontent.com/865226/39891346-46824ad4-546b-11e8-98f5-8172dd035927.png">
I was wondering about that, but when I built the docs, the args in the sig evaluated to their actual values (see screenshot), so I wasn't sure if I needed to call it out...but I guess if they want to change the defaults, they need to know where they are...I am fine with adding that info.

Reply:
yeah, I wonder if it should be in the main text of the docstring rather than blowing up the word count for each of those args.  also wonder if `DEFAULT_FD_OPTIONS` might be more conveniently accessed/reference in `openmdao.api`...

Reply:
@naylor-b @Kenneth-T-Moore any thoughts on @swryan's musings?

Reply:
One thing that will actually change from this is: we currently support a different default stepsize for FD and for CS, but that completely goes away with this pull request, because we explicitly pass in defaults for the stepsize. Maybe the default should go back to None for the default, and we state in the docstring that None means 1e-6 for FD and 1e-15 for CS. I don't see any way to make this convenient for the user (i.e, no "Genetic Optimizer: see Optimizer, Genetic") without hardcoding numbers into the docstring here.